packages/documentation/docs/concepts/accounting/liquidity.md

```diff
@@ -0,0 +1,192 @@
+# Liquidity
+
+## Types of Liquidity
+
+### Asset Liquidity
+
+Asset Liquidity defines the amount of value, denominated in a given asset, Rafiki has at it's disposal to send or forward ILP packets in. It increases if packets denominated in a given asset are received and decreases if packets denominated in a given asset are sent. It is always positive and cannot fall below 0.
+
+Account Servicing Entities should define and adjust the asset liquidity based on their liquidity risk profile.
+
+#### Example
+
+Rafiki has been configured with 2 assets, EUR and USD, both with an asset scale of 0. The EUR liquidity is 10, the USD liquidity if 50.
+
+In a cross-currency transaction, Rafiki receives packets worth 10 EUR and sends packets worth 11 USD. The EUR liquidity is increased to 20, the USD liquidity is reduced to 39.
+
+A transaction where Rafiki receives 50 EUR and would have to sent 55 USD would fail because Rafiki does not have enough USD liquidity.
+
+### Peer Liquidity
+
+Peer Liquidity defines the line of credit, denominated in the asset of the peering relationship, that Rafiki gives a certain peer. It should be defined in the peering agreement and depends on the trust between the transacting peers. Once the peer liquidity is used up, peers should settle and then reset their peer liquidity.
+
+#### Example
+
+A configured peer _Cloud Nine Wallet_ within Rafiki has a peer liquidity of 100 USD. Rafiki can send packets up to 100 USD to payment pointers issued by _Cloud Nine Wallet_. Once that liquidity is used up, we should settle with _Cloud Nine Wallet_ and then reset their liquidity to 100 USD.
+
+### Event Liquidity
+
+When Open Payments incoming or outgoing payments are created, a liquidity account is created within the accounting database. Liquidity needs to be added to an outgoing payment before the payment can be processed. The Account Servicing Entity is notified to add liquidity via the `outgoing_payment.created` event, hence the name _Event Liquidity_. Similarly, packets that are received for an incoming payment increase its liquidity. The Account Servicing Entity is notified to withdraw that liquidity via the `incoming_payment.completed` event.
+
+## Adding and Withdrawing Liquidity
+
+### Asset Liquidity
+
+Add and withdraw asset liquidity via the Admin API (or UI):
+
+```graphql
+mutation AddAssetLiquidity($input: AddAssetLiquidityInput!) {
+  addAssetLiquidity(input: $input) {
+    code
+    success
+    message
+    error
+  }
+}
+```
+
+where
+
+```json
+{
+  "input": {
+    "id": "b97fd85a-126e-42ef-b40d-1a50a70ffa6f",
+    "assetId": "7b8b0f65-896d-4403-b7ba-2e24bf20eb35",
+    "amount": "100",
+    "idempotencyKey": "b97fd85a-126e-42ef-b40d-1a50a70ffa6f"
+  }
+}
+```
```

Comment:
I like the in-context examples because they are clear, but I also wonder about introducing these examples which depend on our current schema/input definitions. I think this is more clear than just linking to the api mutation but now we will need to remember to update these if there are ever any changes. I suppose these are pretty stable but its something to keep the back of our minds.

Reply:
They may not be stable because I just realized that `id` and `idempotencyKey` are basically serving the same purpose and we should drop the `id` from the mutation input. I find those examples to be more informing, too, then linking to the schema. Ideally, we could link to a schema with examples with our new docs framework.

Reply:
Now after out meeting I also wonder, maybe we can link to postman queries? The schema is a nice single source of truth but I do think the usage examples are even clearer.